File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/presentation/dialog/radio_list/RadioDateAdapter.java
package com.bits.bee.bpmc.presentation.dialog.radio_list;

import java.lang.System;

@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000<\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010 \n\u0002\u0010\u000e\n\u0000\n\u0002\u0010\b\n\u0002\b\u000b\n\u0002\u0010\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\b\u0006\u0018\u00002\b\u0012\u0004\u0012\u00020\u00020\u0001:\u0002\u001f B+\u0012\u0006\u0010\u0003\u001a\u00020\u0004\u0012\u0006\u0010\u0005\u001a\u00020\u0006\u0012\f\u0010\u0007\u001a\b\u0012\u0004\u0012\u00020\t0\b\u0012\u0006\u0010\n\u001a\u00020\u000b\u00a2\u0006\u0002\u0010\fJ\u000e\u0010\u000f\u001a\u00020\t2\u0006\u0010\u0010\u001a\u00020\u000bJ\b\u0010\u0011\u001a\u00020\u000bH\u0016J\u0010\u0010\u0012\u001a\u00020\u000b2\u0006\u0010\u0013\u001a\u00020\u000bH\u0016J\u0006\u0010\u0014\u001a\u00020\u000bJ\u0006\u0010\u0015\u001a\u00020\tJ\u0018\u0010\u0016\u001a\u00020\u00172\u0006\u0010\u0018\u001a\u00020\u00022\u0006\u0010\u0013\u001a\u00020\u000bH\u0016J\u0018\u0010\u0019\u001a\u00020\u00022\u0006\u0010\u001a\u001a\u00020\u001b2\u0006\u0010\u001c\u001a\u00020\u000bH\u0016J\u000e\u0010\u001d\u001a\u00020\u00172\u0006\u0010\u001e\u001a\u00020\tR\u000e\u0010\u0003\u001a\u00020\u0004X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0005\u001a\u00020\u0006X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\r\u001a\u00020\tX\u0082.\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u000e\u001a\u00020\u000bX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0014\u0010\u0007\u001a\b\u0012\u0004\u0012\u00020\t0\bX\u0082\u000e\u00a2\u0006\u0002\n\u0000\u00a8\u0006!"}, d2 = {"Lcom/bits/bee/bpmc/presentation/dialog/radio_list/RadioDateAdapter;", "Landroidx/recyclerview/widget/RecyclerView$Adapter;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "childFragmentManager", "Landroidx/fragment/app/FragmentManager;", "ctx", "Landroid/content/Context;", "stringList", "", "", "selected", "", "(Landroidx/fragment/app/FragmentManager;Landroid/content/Context;Ljava/util/List;I)V", "getDateString", "selectedPosition", "getItem", "i", "getItemCount", "getItemViewType", "position", "getSelectedPosition", "getTextDate", "onBindViewHolder", "", "holder", "onCreateViewHolder", "parent", "Landroid/view/ViewGroup;", "viewType", "setTextDate", "toString", "ViewHolderOne", "ViewHolderTwo", "app_debug"})
public final class RadioDateAdapter extends androidx.recyclerview.widget.RecyclerView.Adapter<androidx.recyclerview.widget.RecyclerView.ViewHolder> {
    private final androidx.fragment.app.FragmentManager childFragmentManager = null;
    private final android.content.Context ctx = null;
    private java.util.List<java.lang.String> stringList;
    private int selectedPosition = 0;
    private java.lang.String getDateString;
    
    public RadioDateAdapter(@org.jetbrains.annotations.NotNull()
    androidx.fragment.app.FragmentManager childFragmentManager, @org.jetbrains.annotations.NotNull()
    android.content.Context ctx, @org.jetbrains.annotations.NotNull()
    java.util.List<java.lang.String> stringList, int selected) {
        super();
    }
    
    public final int getSelectedPosition() {
        return 0;
    }
    
    @java.lang.Override()
    public int getItemViewType(int position) {
        return 0;
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public androidx.recyclerview.widget.RecyclerView.ViewHolder onCreateViewHolder(@org.jetbrains.annotations.NotNull()
    android.view.ViewGroup parent, int viewType) {
        return null;
    }
    
    @java.lang.Override()
    public void onBindViewHolder(@org.jetbrains.annotations.NotNull()
    androidx.recyclerview.widget.RecyclerView.ViewHolder holder, int position) {
    }
    
    @java.lang.Override()
    public int getItemCount() {
        return 0;
    }
    
    @org.jetbrains.annotations.NotNull()
    public final java.lang.String getItem(int i) {
        return null;
    }
    
    public final void setTextDate(@org.jetbrains.annotations.NotNull()
    java.lang.String toString) {
    }
    
    @org.jetbrains.annotations.NotNull()
    public final java.lang.String getTextDate() {
        return null;
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u0012\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0004\u0018\u00002\u00020\u0001B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004R\u0011\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\b\n\u0000\u001a\u0004\b\u0005\u0010\u0006\u00a8\u0006\u0007"}, d2 = {"Lcom/bits/bee/bpmc/presentation/dialog/radio_list/RadioDateAdapter$ViewHolderOne;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "binding", "Lcom/bits/bee/bpmc/databinding/ItemRadioButtonBinding;", "(Lcom/bits/bee/bpmc/databinding/ItemRadioButtonBinding;)V", "getBinding", "()Lcom/bits/bee/bpmc/databinding/ItemRadioButtonBinding;", "app_debug"})
    public static final class ViewHolderOne extends androidx.recyclerview.widget.RecyclerView.ViewHolder {
        @org.jetbrains.annotations.NotNull()
        private final com.bits.bee.bpmc.databinding.ItemRadioButtonBinding binding = null;
        
        public ViewHolderOne(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.databinding.ItemRadioButtonBinding binding) {
            super(null);
        }
        
        @org.jetbrains.annotations.NotNull()
        public final com.bits.bee.bpmc.databinding.ItemRadioButtonBinding getBinding() {
            return null;
        }
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u0012\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0004\u0018\u00002\u00020\u0001B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004R\u0011\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\b\n\u0000\u001a\u0004\b\u0005\u0010\u0006\u00a8\u0006\u0007"}, d2 = {"Lcom/bits/bee/bpmc/presentation/dialog/radio_list/RadioDateAdapter$ViewHolderTwo;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "binding", "Lcom/bits/bee/bpmc/databinding/ItemRadioDatepickerBinding;", "(Lcom/bits/bee/bpmc/databinding/ItemRadioDatepickerBinding;)V", "getBinding", "()Lcom/bits/bee/bpmc/databinding/ItemRadioDatepickerBinding;", "app_debug"})
    public static final class ViewHolderTwo extends androidx.recyclerview.widget.RecyclerView.ViewHolder {
        @org.jetbrains.annotations.NotNull()
        private final com.bits.bee.bpmc.databinding.ItemRadioDatepickerBinding binding = null;
        
        public ViewHolderTwo(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.databinding.ItemRadioDatepickerBinding binding) {
            super(null);
        }
        
        @org.jetbrains.annotations.NotNull()
        public final com.bits.bee.bpmc.databinding.ItemRadioDatepickerBinding getBinding() {
            return null;
        }
    }
}

File: app/src/main/java/com/bits/bee/bpmc/presentation/dialog/radio_list/RadioDateAdapter.kt
package com.bits.bee.bpmc.presentation.dialog.radio_list

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.FragmentManager
import androidx.recyclerview.widget.RecyclerView
import com.bits.bee.bpmc.databinding.ItemRadioButtonBinding
import com.bits.bee.bpmc.databinding.ItemRadioDatepickerBinding
import com.bits.bee.bpmc.utils.BPMConstants
import com.bits.bee.bpmc.utils.DateFormatUtils
import com.google.android.material.datepicker.MaterialDatePicker

class RadioDateAdapter(
    private val childFragmentManager: FragmentManager,
    private val ctx: Context,
    stringList: List<String>,
    selected: Int
): RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private var stringList : List<String> = mutableListOf()
    private var selectedPosition : Int = 0
    private var getDateString: String? = null

    init {
        this.stringList = stringList
        this.selectedPosition = selected
    }

    fun getSelectedPosition() : Int = selectedPosition

    override fun getItemViewType(position: Int): Int {
        if (!stringList.get(position).contains("Custom")){
            return 0
        }
        return 1
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        if (viewType == 0){
            return ViewHolderOne(ItemRadioButtonBinding.inflate(inflater, parent, false))
        }
        return ViewHolderTwo(ItemRadioDatepickerBinding.inflate(inflater, parent, false))
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (!stringList.get(position).contains("Custom")){
            val holderOne: ViewHolderOne = holder as ViewHolderOne
            holderOne.binding.apply {
                materialRadioButton.text = getItem(holder.absoluteAdapterPosition)

                materialRadioButton.isChecked = selectedPosition == holder.absoluteAdapterPosition

                materialRadioButton.setOnClickListener {
                    selectedPosition = holder.absoluteAdapterPosition
                    notifyDataSetChanged()
                }
            }
        }else{
            val holderTwo: ViewHolderTwo = holder as ViewHolderTwo
            holderTwo.binding.apply {
                materialRadioButton.text = getItem(holder.absoluteAdapterPosition)
                materialRadioButton.isChecked = selectedPosition == holder.absoluteAdapterPosition

                materialRadioButton.setOnClickListener {
                    selectedPosition = holder.absoluteAdapterPosition
                    notifyDataSetChanged()
                }

                if (materialRadioButton.isChecked){
                    etPilihTgl.visibility = View.VISIBLE
                }else{
                    etPilihTgl.visibility = View.GONE
                }

                etPilihTgl.setOnClickListener {
                    val datePicker = MaterialDatePicker.Builder.dateRangePicker().build()
                    datePicker.show(childFragmentManager, "")

                    datePicker.addOnPositiveButtonClickListener { dateSelected ->
                        val startDate = dateSelected.first
                        val endDate = dateSelected.second

                        etPilihTgl.setText("${DateFormatUtils.convertLongToTime(BPMConstants.NEW_DATE_FORMAT, startDate)} - " +
                                "${DateFormatUtils.convertLongToTime(BPMConstants.NEW_DATE_FORMAT, endDate)}")
                        setTextDate(holderTwo.binding.etPilihTgl.text.toString())
                    }

                }

            }
        }
    }

    override fun getItemCount(): Int {
        return stringList.size
    }


    fun getItem(i : Int) = stringList[i]

    class ViewHolderOne(val binding : ItemRadioButtonBinding) : RecyclerView.ViewHolder(binding.root)

    class ViewHolderTwo(val binding : ItemRadioDatepickerBinding) : RecyclerView.ViewHolder(binding.root)

    fun setTextDate(toString: String){
        this.getDateString = toString
    }

    fun getTextDate(): String{
        return getDateString ?: ""
    }



}